PowerEditor/src/Notepad_plus.cpp

```diff
@@ -6283,6 +6279,20 @@ void Notepad_plus::setWorkingDir(const TCHAR *dir)
 	}
 }
 
+void Notepad_plus::setWorkingDir(const Buffer* buf)
+{
+	if (buf)
+	{
+		generic_string dir(buf->getFullPathName());
+		PathRemoveFileSpec(dir);
+		setWorkingDir(dir.c_str());
+	}
+	else
+	{
+		setWorkingDir(static_cast<const TCHAR*>(nullptr));

```

Comment:
Why do we need to add this method, whereas `setWorkingDir(TCHAR*)` is called inside ?